src/service_client/assisted_service_api.py

```diff
@@ -323,7 +323,7 @@ def download_and_save_infra_env_file(self, infra_env_id: str, file_name: str, fi
         response = self.client.v2_download_infra_env_files(
             infra_env_id=infra_env_id, file_name=file_name, _preload_content=False
         )
-        with open(file_path, "wb") as _file:
+        with open(file_path+file_name, "wb") as _file:
```

Comment:
this is changing the behavior of this function, have you checked the usage of it? In order to be safe, I would suggest to call it with `file_path` representing the destination file (see my suggestion above).